src/listpack.c

```diff
@@ -1027,24 +1027,18 @@ unsigned char *lpDeleteRange(unsigned char *lp, long index, unsigned long num) {
     if (num == 0) return lp; /* Nothing to delete, return ASAP. */
     if ((p = lpSeek(lp, index)) == NULL) return lp;
 
-    if (numele == LP_HDR_NUMELE_UNKNOWN) {
-        /* If the listpack length cannot be obtained in constant time,
-         * using lpDeleteRangeWithEntry will be much faster. */
-        lp = lpDeleteRangeWithEntry(lp, &p, num);
+    /* If we know we're gonna delete beyond the end of the listpack, we can just move
+     * the EOF marker, and there's no need to iterate through the entries,
+     * but if we can't be sure how many entries there are, we rather avoid calling lpLength
+     * since that means an additional iteration on all elements. */
+    if (num != LP_HDR_NUMELE_UNKNOWN && index < 0) index = (long)numele + index;
+    if (num != LP_HDR_NUMELE_UNKNOWN && (numele - (unsigned long)index) <= num) {
+        p[0] = LP_EOF;
+        lpSetTotalBytes(lp, p - lp + 1);
+        lpSetNumElements(lp, index);
+        lp = lpShrinkToFit(lp);
     } else {
-        /* Note that index could overflow, but we use the value
-        * after seek, so when we use it no overflow happens. */
```

Comment:
you may still wanna keep this comment if you think it's useful (personally i didn't dive into it)

Reply:
I thought no one would care about this comment, I think it's necessary (I often confusing).

Reply:
I still need to do the benchmark.

Reply:
I also need to run "corrupt-dump-fuzzer " for several hours.